sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/implementation/EventHubSharedKeyCredential.java

```diff
@@ -153,4 +180,22 @@ private AccessToken generateSharedAccessSignature(final String resource) throws
 
         return new AccessToken(token, expiresOn);
     }
+
+    private OffsetDateTime getExpirationTime(String sharedAccessSignature) {
+        String[] parts = sharedAccessSignature.split("&");
+        return Arrays.stream(parts)
+            .map(part -> part.split("="))
+            .filter(pair -> pair.length == 2 && pair[0].equalsIgnoreCase("se"))
+            .findFirst()
+            .map(pair -> pair[1])
+            .map(expirationTimeStr -> {
+                try {
+                    long epochSeconds = Long.parseLong(expirationTimeStr);
+                    return new Date(epochSeconds * 1000).toInstant().atOffset(ZoneOffset.UTC);
+                } catch (NumberFormatException exception) {
+                    return OffsetDateTime.MAX;
```

Comment:
Should we log this? It would be nice to understand why we returned offsetdatemax in case there was some incorrect formatting in the `se`.